presubmit-service/check_if_pr_can_cherry_pick.sh

```diff
@@ -0,0 +1,63 @@
+#! /bin/bash
+
+# The organization containing the repo to be checked against, e.g. angular
+ORG=$1
+# The repo to be checked against, e.g. material2
+REPO=$2
+# The PR number to retrieve the patch files for, e.g. 1234
+PR=$3;
+# The branch in the repo to check for cherry-picking onto, e.g. 6.1.x
+BRANCH=$4;
+# The fullname of the repository to be checked against, e.g. angular/material2
+FULL_REPO="$ORG/$REPO"
+# Whether the PR can be cherry-picked on to the targetted branch.
+RESULT=false;
+
+# If the the repos directory does not exist, create it.
+[ ! -d "local_repo" ] && mkdir local_repo;
+cd local_repo;
+
+# If the current directory, which should be repos is not a toplevel of a git repo
+# initialize a repo and create a dummy commit in to create a local remote with a
+# master branch.
+if [ "$(git rev-parse --show-toplevel)" != "$(pwd)" ]; then
+  git init &>/dev/null;
+  touch empty_repo &>/dev/null;
+  git add * &>/dev/null;
+  git commit -a -m "empty repo" &>/dev/null;
+fi
+
+# If the repo does not exist in the list of repos for the local git repo, add it
+# to the remotes.
+git remote get-url $FULL_REPO  &>/dev/null;
+if [ $? -ne 0 ]; then
+  git remote add $FULL_REPO "https://github.com/$FULL_REPO.git" &>/dev/null;
+fi
+
+# Fetch for the targeted repo.
+git fetch --depth 1 $FULL_REPO &>/dev/null;
+
+# Retrieve the patch file for the PR from github.
+curl -L -s https://github.com/$FULL_REPO/pull/$PR.patch > /tmp/$PR.patch;
```

Comment:
How about having a different directory for the repository slug? `/tmp/presubmit-service/$FULL_REPO/4343.patch` because otherwise if this is being used for multiple repositories, PR numbers might conflict. (considering that URLs can run concurrently)

Also it's just best practice to not pollute `/tmp` by having a namespace in the temp directory I think.